### PR TITLE
Take the Threads Activity Centre out of labs

### PR DESCRIPTION
**Pull request by dbkr**
Requires https://github.com/matrix-org/matrix-react-sdk/pull/12438 and ideally https://github.com/matrix-org/matrix-react-sdk/pull/12418

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

**Comment by dbkr** [comment]
This now has two end to end tests that are still failing. They appear to be legitimate failures in the app in so far as the activity dot re-appears on a thread after a redaction and a refresh. I tracked it down as far as establishing that the redacted event and redaction are fetched from the saved sync, which they'll have to be in order for this to work at all, or we simply won't know about the event our read receipt points to. Some time subsequently, the redaction appears to be 'processed', removing both the redaction and the redacted event from the timeline, so we're now in the state we'd be in if we hadn't synced from the cache, therefore the unread dot appears.

I'm not sure how this was working in the old code. The test passes when I run it locally but not when I debug it. I suspect it's been passing by some fluke and is actually at least partially broken.

The best thing to do here might be to disable this test for now and look at it when we have time to debug it. The alternative is blocking the TAC behind fixing this.

File: test/components/structures/MatrixChat-test.tsx [hunk 1]
```diff
@@ -676,7 +683,8 @@ describe("<MatrixChat />", () => {
                         it("should warn when room is not public", async () => {
                             jest.spyOn(room.currentState, "getStateEvents").mockReturnValue(inviteJoinRule);
                             dispatchAction();
-                            await screen.findByRole("dialog");
+                            const blerp = await screen.findByRole("dialog");
+                            console.log(blerp);
```
Comment (t3chguy) [inline test/components/structures/MatrixChat-test.tsx:687]: ?

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: test/components/structures/MatrixChat-test.tsx (hunk 1 above)
Comment (dbkr) [reply test/components/structures/MatrixChat-test.tsx]: Blerp! Blerp, blerp. (Oops! Sorry, fixed.)

File: test/components/structures/MatrixChat-test.tsx [hunk 2]
```diff
@@ -676,7 +683,6 @@ describe("<MatrixChat />", () => {
                         it("should warn when room is not public", async () => {
                             jest.spyOn(room.currentState, "getStateEvents").mockReturnValue(inviteJoinRule);
                             dispatchAction();
-                            await screen.findByRole("dialog");
```
Comment (t3chguy) [inline test/components/structures/MatrixChat-test.tsx:679]: I'm not sure removing this is expected, likely will make the test racey

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: test/components/structures/MatrixChat-test.tsx (hunk 2 above)
Comment (dbkr) [reply test/components/structures/MatrixChat-test.tsx]: ah yes, in fact it just made it fail outright

**Review by t3chguy — APPROVED** [review]
LGTM assuming a Product tick

**Comment by dbkr** [comment]
(Danielle says yes)

**Review by daniellekirkwood — COMMENTED** [review]
Product happy :)

**Pull request merged**